src/model/Material.hpp

```diff
@@ -95,6 +101,25 @@ namespace model {
     // resets the material property moisture penetration depth settings
     void resetMaterialPropertyMoisturePenetrationDepthSettings();
 
+    // if material property phase change already exists, do nothing and return nil; creates the material property phase change if it does not already exist and return it
+    boost::optional<MaterialPropertyPhaseChange> createMaterialPropertyPhaseChange();
```

Comment:
Just a note: I know you're just following the existing construct (`createMaterialPropertyMoisturePenetrationDepthSettings`) but ideally we should have called them something more explicit. Like `getOrCreateXXX`

Reply:
🤷‍♂️ . It's just a convenience on the Material object. You can also call the ctor explicitly and pass Material in...